.gitignore

```diff
@@ -5,6 +5,9 @@
 *.so
 *.dylib
 
+# os generated files
+.DS_Store
```

Comment:
Fine to add this but you'll never have to do it again if you run `mkdir -p ~/.config/git && echo .DS_Store >> ~/.config/git/ignore`.